include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
@jfirebaugh I Fixed up ownership as discussed above in https://github.com/mapbox/mapbox-gl-native/pull/8797/commits/7f0ad3241db420e8a91242f1af7586592aa0ce47. I Left the pointers to the concrete Implementations in the subclasses as is (`const style::FooLayer::Impl* const impl;` as opposed to a `const style::FooLayer::Impl&`) to avoid modifying all call sites.